src/backend/vulkan/src/lib.rs

```diff
@@ -353,6 +353,7 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
     unsafe fn open(
         &self,
         families: &[(&QueueFamily, &[hal::QueuePriority])],
+        requested_features: Option<Features>,
```

Comment:
let's assert that requested features are a subset of available features

Reply:
I added an assertion for the Vulkan backend, should the same be done for the other backends as well?

Reply:
On second thought actually, wouldn't it be better to just return an error rather than panicking? We already have the `MissingFeature` error anyway.